CARDS-1404: As a questionnaire designer I can specify numeric questions of type "range"

marta- opened the pull request:
* Made `isRange` a question definition field and enabled it in the questionnaire wizard
* Fixed bug where the upper limit of the range was dropped on form re-editing
* Improved UI of range input fields (width, labels, placeholders, alignment)
* Improved how ranges are displayed in view mode, on the form and in the subject chart
* Improved error reporting
* Improved variable names
* Added a `Range Field Test` questionnaire in the `test` run mode

**NOT done**:
* [ ] Flag range answer as invalid if lower limit > upper limit
* [ ] Generate range display value in JSON

marta- committed:
CARDS-1404: As a questionnaire designer I can specify numeric questions of type "range"
* Made `isRange` a question definition field and enabled it in the questionnaire wizard
* Fixed bug where the upper limit of the range was dropped on form re-editing
* Improved UI of range input fields (width, labels, placeholders, alignment)
* Improved how ranges are displayed in view mode, on the form and in the subject chart
* Improved error reporting
* Improved variable names
* Added a `Range Field Test` questionnaire in the `test` run mode

NOT done:
* Flag range answer as invalid if lower limit > upper limit
* Generate range display value in JSON

sashaandjic reviewed (approved):
tested

marta- committed:
CARDS-1404: As a questionnaire designer I can specify numeric questions of type "range"
Addressing review comments: use `helperText` instead of `label` to mark the range inputs for the user as "Lower limit" and "Upper limit", for consistency with other question types.

sashaandjic reviewed (approved):
re-tested

sdumitriu reviewed (approved):
Minor bugs in Firefox:
- tabbing between range inputs requires two tabs; the first tab selects the `TextField` which cannot be edited, and the second tab actually goes into the `Input`; I think the `helperText` adds this behavior, but I didn't test without it
- entering a very long number causes the helper text to scroll out of view; it would be best if it stayed in place
- actually, typing a very long number causes the cursor to go out of view, pressing `End` to scroll to the end will bring the cursor back into view and scroll the helper text out of view
- typing a long number, jumping to the `End`, then deleting with `Backspace` will cause the cursor to go out of view again, this time to the left, and typing will not be visible

The cursor bugs are also somehow occuring in Chrome, but with slightly better behavior.

marta- committed:
CARDS-1404: As a questionnaire designer I can specify numeric questions of type "range"
Addressing review comments: fixed the width of the TextFields, to address issues with helper text and cursor visibility

marta- commented:
> Minor bugs in Firefox:
> 
> * tabbing between range inputs requires two tabs; the first tab selects the `TextField` which cannot be edited, and the second tab actually goes into the `Input`; I think the `helperText` adds this behavior, but I didn't test without it

I can't replicate the tabbing issue.

> * entering a very long number causes the helper text to scroll out of view; it would be best if it stayed in place
> * actually, typing a very long number causes the cursor to go out of view, pressing `End` to scroll to the end will bring the cursor back into view and scroll the helper text out of view
> * typing a long number, jumping to the `End`, then deleting with `Backspace` will cause the cursor to go out of view again, this time to the left, and typing will not be visible
> 
> The cursor bugs are also somehow occuring in Chrome, but with slightly better behavior.

Issues that appear when entering a long number are addressed in 13512f7 .
